fmriprep/workflows/base.py

```diff
@@ -429,10 +429,20 @@ def init_single_subject_wf(subject_id: str):
 
             fmap_estimators = [fmap for fmap in fmap_estimators if fmap.bids_id in used_estimators]
 
+            # Simplification: Unused estimators are removed from registry
+            # This fiddles with a private attribute, so it may break in future
+            # versions. However, it does mean the BOLD workflow doesn't need to
+            # replicate the logic that got us to the pared down set of estimators
+            # here.
+            final_ids = {fmap.bids_id for fmap in fmap_estimators}
+            unused_ids = [bids_id for bids_id in fm._estimators if bids_id not in final_ids]
```

Comment:
Now that I'm thinking of it as a set, we might as well do:

```suggestion
            unused_ids = fm._estimators.keys() - final_ids
```

From https://docs.python.org/3.8/library/stdtypes.html#dict-views

> Keys views are set-like since their entries are unique and hashable. If all values are hashable, so that (key, value) pairs are unique and hashable, then the items view is also set-like. (Values views are not treated as set-like since the entries are generally not unique.) For set-like views, all of the operations defined for the abstract base class [collections.abc.Set](https://docs.python.org/3.8/library/collections.abc.html#collections.abc.Set) are available (for example, ==, <, or ^).